src/pages/ShowResource.tsx

```diff
@@ -0,0 +1,51 @@
+import { Typography } from "@material-ui/core";
+import React from "react";
+import { useAsync } from "react-async";
+import { RouteComponentProps } from "react-router-dom";
+import getComitResource from "../api/getComitResource";
+import { GetSwapResponse } from "../api/swapResource";
+import { GetSwapsResponse } from "../api/swapsResource";
+import CenteredProgress from "../components/CenteredProgress";
+import ErrorSnackbar from "../components/ErrorSnackbar";
+import SwapList from "./ListSwaps/ListSwaps";
+import Swap from "./SwapPage/Swap";
+
+const getComitResourceFn = async ({ resourcePath }: any) => {
+  return getComitResource(resourcePath);
+};
+
+function ShowResource({ location }: RouteComponentProps) {
+  const resourcePath = location.pathname.replace("/show_resource/", "");
+
+  const { data, isLoading, error } = useAsync({
+    promiseFn: getComitResourceFn,
+    resourcePath,
+    /* FIXME: If connection lost whilst looking at a swap page and then
+       try to navigate back to swap list, nothing happens */
+    watch: location.pathname
```

Comment:
IMO, the hook should re-trigger if its arguments change. If that is not the case, then we need to drop this library :sweat_smile:

Reply:
But maybe this is related to the way the routing is set up?

Reply:
I need to investigate this some more, but I think the library is fine.

Reply:
This was fixed, btw. The hook was triggered because of the change in the path, it failed to fetch, but the data object didn't change. The problem was with the way I ordered the if statements afterwards.